### PR TITLE
[3D] Add tests for Qgscameracontroller

### PR DESCRIPTION
**Pull request by ptitjano**

### CODE REVIEW

File: src/3d/qgscameracontroller.h [hunk 1]
```diff
@@ -48,7 +48,8 @@ class QgsCameraPose;
 class QgsVector3D;
 class QgsWindow3DEngine;
 class Qgs3DMapScene;
-class QgsCameraController4Test;
+class TestQgs3DRendering;
```
Comment (nyalldawson) [inline src/3d/qgscameracontroller.h:51]: Thanks! You can safely drop these forward declarations, they aren't required for friend classes .

Comment (ptitjano) [reply src/3d/qgscameracontroller.h]: Compilation fails when I drop these forward declarations. Am I missing something?

Comment (ptitjano) [reply src/3d/qgscameracontroller.h]: My bad, I was using `friend TestQgs3DRendering;` instead of `friend class TestQgs3DRendering;`. This is now fixed.

**Review by nyalldawson — APPROVED** [review]

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
@ptitjano these new tests are randomly failing, see eg https://github.com/qgis/QGIS/actions/runs/8906456520/job/24463412554?pr=57295

**Pull request merged**